internal/core/src/rescores/Scorer.h

```diff
@@ -16,16 +16,49 @@
 
 #pragma once
 
+#include <exception>
+#include "common/EasyAssert.h"
+#include "common/Types.h"
 #include "expr/ITypeExpr.h"
+#include "pb/common.pb.h"
+#include "pb/plan.pb.h"
+#include "segcore/SegmentInterface.h"
+#include "common/protobuf_utils.h"
 
 namespace milvus::rescores {
 class Scorer {
  public:
+    virtual ~Scorer() = default;
+
     virtual expr::TypedExprPtr
     filter() = 0;
 
-    virtual float
-    rescore(float old_score) = 0;
+    // filter result of offset[i] was bitmapview[i]
+    // add boost score for idx[i] if bitmap[i] was true
+    virtual void
+    batch_score(const segcore::SegmentInternalInterface* segment,
+                const proto::plan::FunctionMode& mode,
+                const FixedVector<int32_t>& offsets,
+                const TargetBitmapView& bitmap,
```

Comment:
how about template `TargetBitmapView/TargetBitmap` to avoid duplicate code?

Reply:
Not duplicate code. Here was different behavior on Bitmap and BitmapView.